test/test_frangel.jl

```diff
@@ -0,0 +1,88 @@
+using Herb.HerbGrammar: @cfgrammar, rulenode2expr
+using Herb.HerbSearch: BFSIterator
+using Herb.HerbSpecification: IOExample, Problem
+using Herb.HerbCore: RuleNode
+using Garden: mine_fragments, select_shallowest_fragments, select_smallest_fragments, modify_grammar_frangel!, decide_frangel, NoProgramFoundError
+
+@testset verbose=true "FrAngel" begin
+    @testset "Integration tests" begin
+        # Define extra grammar as FrAngel will change it.
+        grammar = @cfgrammar begin
+            Start = Int
+            Int = Int + Int
+            Int = 1 | 2
+        end
+        problem = Problem(
+            [IOExample{Symbol, Any}(Dict(), 2)]
+        )
+        result = frangel(
+            BFSIterator,
+            grammar,
+            :Start,
+            problem;
+            max_depth=4
+        )
+
+        @test rulenode2expr(result,grammar) == 2
+
+        imp_problem = Problem(
+            [IOExample{Symbol, Any}(Dict(), 0)]
+        )
+ 
+        # A program yielding 0 is impossible to derive from the grammar.
+        @test_throws NoProgramFoundError frangel(BFSIterator, grammar, :Start, imp_problem; max_depth=4)
+    end
+
+    grammar = @cfgrammar begin
+        Start = Int
+        Int = Int + Int
+        Int = 1 | 2
+    end
+
+    @testset "mine_fragments" begin
+        rn = RuleNode(2,
+            [RuleNode(2,[RuleNode(3),RuleNode(4)]),RuleNode(3)]
+        ) # 2{2{3,4},3}
+
+        fragments = collect(mine_fragments(grammar, rn))
```

Comment:
```suggestion
        fragments = mine_fragments(grammar, rn)
```

Can avoid converting this to a vector if we just use membership testing below.